packages/zoe/src/zoe.js

```diff
@@ -24,10 +25,215 @@ import { areRightsConserved } from './rightsConservation';
+ * with keywords as keys. Terms are the arguments to the contract,
+ * such as the number of bids an auction will wait for before closing.
+ * Terms are up to the discretion of the smart contract. We get back
+ * an invite (an ERTP payment) to participate in the contract.
+ *
+ * TODO: property {(instanceHandle: InstanceHandle) => InstanceRecord} getInstanceRecord
+ * Credibly get information about the instance (such as the installation
+ * and terms used).
+ *
+ * @property {(invite: Invite, proposal: Proposal, payments: PaymentKeywordRecord) => Promise<SeatAndPayout>} redeem
+ * To redeem an invite, the user normally provides a proposal (their rules for the
+ * offer) as well as payments to be escrowed by Zoe.  If either the proposal or payments
+ * would be empty, indicate this by omitting that argument or passing undefined, rather
+ * than passing an empty record.
+ *
+ * The proposal has three parts: `want` and `give` are used
+ * by Zoe to enforce offer safety, and `exit` is used to specify
+ * the extent of payout liveness that Zoe can guarantee.
+ * `want` and `give` are objects with keywords as keys and amounts
+ * as values. `payments` is a record with keywords as keys,
+ * and the values are the actual payments to be escrowed. A payment
+ * is expected for every rule under `give`.
+ *
+ * @property {(installationHandle: InstallationHandle) => InstallationRecord} getInstallation
+ *
+ * @typedef {Object} SeatAndPayout This is returned by a call to `redeem` on Zoe.
+ *
+ * @property {Object} seat An arbitrary object whose methods allow the user to take
+ * certain actions in a contract. The payout is a promise that resolves
+ * to an object which has keywords as keys and promises for payments
+ * as values. Note that while the payout promise resolves when an offer
+ * is completed, the promise for each payment resolves after the remote
+ * issuer successfully withdraws the payment.
+ *
+ * @property {Payment[]} payout A promise that resolves
+ * to an object which has keywords as keys and promises for payments
+ * as values. Note that while the payout promise resolves when an offer
+ * is completed, the promise for each payment resolves after the remote
+ * issuer successfully withdraws the payment.
+ *
+ * @typedef {Object} Proposal
+ * @property {AmountKeywordRecord} want
+ * @property {AmountKeywordRecord} give
+ * @property {ExitRule} exit
+ *
+ * @typedef {Object.<string,Amount>} AmountKeywordRecord
+ * The keys are keywords, and the values are amounts. For example:
+ * { Asset: amountMath.make(5), Price: amountMath.make(9) }
+ *
+ * @typedef {Object} ExitRule
+ * The possible keys are 'waived', 'onDemand', and 'afterDeadline'.
+ * `timer` and `deadline` only are used for the `afterDeadline` key.
+ * The possible records are:
+ * `{ waived: null }`
+ * `{ onDemand: null }`
+ * `{ afterDeadline: { timer :Timer<Deadline>, deadline :Deadline } }
+ * @property {Timer} [timer]
+ * @property {Deadline} [deadline]
```

Comment:
`Deadline` doesn't seem to be defined. Should it be `number` instead?

Reply:
This was inherited from the chainmail.
```
* @typedef {number} Deadline XXX What is the unit for this?
```
would be a fine start.  You fill out the `XXX` :wink: